stan/math/prim/mat/prob/ordered_logistic_glm_lpmf.hpp

```diff
@@ -0,0 +1,222 @@
+#ifndef STAN_MATH_PRIM_MAT_PROB_ORDERED_LOGISTIC_GLM_LPMF_HPP
+#define STAN_MATH_PRIM_MAT_PROB_ORDERED_LOGISTIC_GLM_LPMF_HPP
+
+#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
```

Comment:
Please replaces the meta includes with prim/meta.hpp, rev/meta.hpp etc. We should really test for these with test-math-dependencies or something like that.

Reply:
The reason to include higher level is to get the order of includes right among the generic, `std::vector`, and `Eigen::Matrix` signatures. 

If we only include **`prim`**/`meta.hpp`, will the higher level `rev/meta.hpp` get included in the right order elsewhere if any `rev` gets included?

Reply:
@bob-carpenter: the short answer is yes. If this file is in prim, it should include the prim meta file. When it gets used with rev, the client of the Math library should be including rev/mat.hpp, which includes rev/meta.hpp prior to including this file.